app/services/marketplace_service/store/search_settings.rb

```diff
@@ -0,0 +1,49 @@
+module MarketplaceService::Store::SearchSettings
+
+  SearchSettingsModel = ::SearchSettings
+
+  SearchSettings = EntityUtils.define_builder(
+    [:community_id, :mandatory, :fixnum],
+    [:main_search,:to_symbol, :mandatory, one_of: [:keyword, :location]]
+  )
+
+  module_function
+
+  def create(opts)
+    settings = HashUtils.compact(SearchSettings.call(opts))
+    model = SearchSettingsModel.create!(settings)
+    from_model(model)
+  end
+
+  def update(opts)
+    settings = SearchSettings.call(opts)
+
+    Maybe(find(opts[:community_id]))
+      .map { |model|
+        model.update_attributes!(settings)
+        from_model(model)
+      }
+      .or_else(nil)
+  end
+
+  def get(community_id:)
+    Maybe(find(community_id))
+      .map { |m| from_model(m) }
+      .or_else(nil)
+  end
+
+
+  ## Privates
+
+  def from_model(model)
+    Maybe(model)
+      .map { |m| EntityUtils.model_to_hash(m) }
+      .map { |hash| SearchSettings.call(hash) }
+      .or_else(nil)
+  end
+
+  def find(community_id)
+    SearchSettingsModel.where(community_id: community_id).first
```

Comment:
New Rails 4 thing! You can also use `SearchSettingsModel.find_by(community_id: community_id)`